frontend/src/views/DocumentListView.vue

```diff
@@ -0,0 +1,150 @@
+import Button from "primevue/button";
+import Column from "primevue/column";
+import DataTable from "primevue/datatable";
+import InputText from "primevue/inputtext";
+import { onMounted, ref, watch } from "vue";
+
+import { deleteDocument, getDocumentPdf, getDocuments } from "@/backendClient";
+import useConfirmations from "@/compositions/useConfirmations";
+import useNotifications from "@/compositions/useNotifications";
+import { DocumentKind } from "@/global/types/appTypes";
+import type {
+  InvoiceDocument,
+  OfferDocument,
+  OverdueNoticeDocument,
+} from "@/global/types/entities";
+import { debounce } from "@/helpers/utils";
+import { ROUTES } from "@/router";
+
+const confirm = useConfirmations();
+const notifications = useNotifications();
+const documentsList = ref<(InvoiceDocument | OfferDocument | OverdueNoticeDocument)[]>([]);
+const search = ref<string>("");
+const selectedProduct = ref<(OfferDocument | OverdueNoticeDocument | InvoiceDocument)[]>([]);
+
+function getDocumentType(doc: OfferDocument | OverdueNoticeDocument | InvoiceDocument) {
+  if ("offer_id" in doc) {
+    return DocumentKind.offer;
+  } else if ("overdue_notice_id" in doc) {
+    return DocumentKind.overdueNotice;
+  } else {
+    return DocumentKind.invoice;
+  }
+}
+
+async function reloadPage() {
+  // TODO: use pagination
+  documentsList.value = await getDocuments(search.value);
+}
+
+async function removeDocument(doc: OfferDocument | OverdueNoticeDocument | InvoiceDocument) {
+  let kind = getDocumentType(doc);
+  await deleteDocument(doc.id, kind);
+  reloadPage();
+}
+
+function getSelectedIds() {
+  return selectedProduct.value.map(function (element) {
+    return element.id;
+  });
+}
+
+async function createPdf() {
+  let selectedValuesArray = selectedProduct.value.map(function (element) {
+    return { kind: getDocumentType(element), id: element.id };
+  });
+  const response = await getDocumentPdf(selectedValuesArray);
+  notifications.showCreatePdfDocumentNotification();
+  saveAs(response, `${getSelectedIds().join("-")}.pdf`);
+}
+
+const confirmCreatePdf = () => {
+  confirm.showCreatePdfConfirmation(createPdf, getSelectedIds());
+};
+
+watch(search, debounce(reloadPage, 250));
+onMounted(async () => {
+  reloadPage();
+});
+</script>
+
+<template>
+  <div class="flex flex-col overflow-hidden h-full">
+    <div class="card flex w-full gap-x-6 mb-3">
+      <span class="relative grow">
+        <i
+          class="pi pi-search absolute top-2/4 -mt-2 left-3 text-surface-400 dark:text-surface-600"
+        />
+        <InputText
+          v-model="search"
+          placeholder="Suche"
+          class="pl-10 w-full"
+          data-testid="client-search-input"
+        />
+      </span>
+      <Button
+        @click="confirmCreatePdf"
+        type="button"
+        label="Pdf Speichern"
+        :disabled="!selectedProduct.length"
+      />
+    </div>
+    <div v-if="documentsList" class="grow overflow-auto">
+      <DataTable
```

Comment:
This is the first place we use a DataTable. How come?

Reply:
We need to use here Datatable because we have a checkbox, in other places we didn't have it. Do you want to use it for the other list views?

Reply:
I'll compare the different approaches later a bit and see if it makes sense to harmonize them 👌🏻